common/src/main/java/com/microsoft/identity/common/internal/telemetry/AndroidTelemetryContext.java

```diff
@@ -67,9 +68,13 @@ private void addApplicationInfo(@NonNull final Context context) {
             String packageName = "";
 
             if (applicationInfo != null) {
-                applicationName = applicationInfo.labelRes == 0 && applicationInfo.nonLocalizedLabel != null ?
-                        applicationInfo.nonLocalizedLabel.toString() : context.getString(applicationInfo.labelRes);
                 packageName = applicationInfo.packageName;
+                try {
+                    applicationName = context.getString(applicationInfo.labelRes);
+                } catch (Resources.NotFoundException exception) {
```

Comment:
What would actually cause to happen? Can it be prevented altogether?

Reply:
I was actually asking about the new code here:

context.getString(applicationInfo.labelRes);

Seems like this throws Resources.NotFoundException
In what cases does it through that exception? I mean how likely are we to hit this exception in our production code?

Reply:
This cannot happen in production since there is an application label in the application context. 
It will only fail for our robolectric tests.

Reply:
Okay, I think we should still keep the try/catch block and log the error when they occur